modules/standard/RecordParser.chpl

```diff
@@ -17,45 +17,136 @@
  * limitations under the License.
  */
 
-use IO, Regexp;
 
-/* A general purpose record reader/parser for channels.
+/*
+
+A general purpose record reader/parser for channels. Uses a regular expression
+to capture portions of the input, and then assigns each capture to each
+record field.
+
+This module makes the following assumptions:
+
+1. The file has been opened for reading and a reader channel has been created
+
+2. The programmer has created a record (NOT a class) that can have an arbitrary
+   number of fields that can be cast from strings (i.e no subrecords, arrays
+   etc).
```

Comment:
This sentence feels a little awkward to me.  Perhaps "a record ... of an arbitrary number of fields that can be cast" instead?